### ISSUE TITLE
Auctionmark NumberFormat Exception

### ISSUE THREAD

**Issue by pct960** [cmu-db/benchbase#335]
Hi everyone,

I'm trying to run the AuctionMark benchmark on Postgres, and I'm getting an unexpected NumberFormatException. I tried to run the test for 900s at 2k TPS with sampling enabled every second. The test threw a whole bunch of warnings about duplicate items before finally crashing at around the 500th second. The log output is below:

```
.
.
.
[WARN ] 2023-07-04 00:01:50,948 [AuctionMarkWorker<000>]  com.oltpbenchmark.benchmarks.auctionmark.AuctionMarkWorker executeNewItem - a duplicate item existed; i believe this error should be handled differently.: Duplicate ItemId [000000000200000005330000000002] for Seller [00000000020000000533]
[WARN ] 2023-07-04 00:01:51,133 [AuctionMarkWorker<000>]  com.oltpbenchmark.benchmarks.auctionmark.AuctionMarkWorker executeNewItem - a duplicate item existed; i believe this error should be handled differently.: Duplicate ItemId [000000000100000006640000000001] for Seller [00000000010000000664]
[WARN ] 2023-07-04 00:01:51,455 [AuctionMarkWorker<000>]  com.oltpbenchmark.benchmarks.auctionmark.AuctionMarkWorker executeNewItem - a duplicate item existed; i believe this error should be handled differently.: Duplicate ItemId [000000000100000013440000000001] for Seller [00000000010000001344]
[WARN ] 2023-07-04 00:01:51,463 [AuctionMarkWorker<000>]  com.oltpbenchmark.benchmarks.auctionmark.AuctionMarkWorker executeNewItem - a duplicate item existed; i believe this error should be handled differently.: Duplicate ItemId [000000000200000001510000000002] for Seller [00000000020000000151]
[WARN ] 2023-07-04 00:01:51,769 [AuctionMarkWorker<000>]  com.oltpbenchmark.benchmarks.auctionmark.AuctionMarkWorker executeNewItem - a duplicate item existed; i believe this error should be handled differently.: Duplicate ItemId [000000000400000001870000000004] for Seller [00000000040000000187]
[WARN ] 2023-07-04 00:01:51,822 [AuctionMarkWorker<000>]  com.oltpbenchmark.benchmarks.auctionmark.AuctionMarkWorker executeNewItem - a duplicate item existed; i believe this error should be handled differently.: Duplicate ItemId [000000000100000010860000000001] for Seller [00000000010000001086]
[ERROR] 2023-07-04 00:01:51,967 [AuctionMarkWorker<000>]  com.oltpbenchmark.ThreadBench uncaughtException - For input string: "0-67108864"
java.lang.NumberFormatException: For input string: "0-67108864"
        at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
        at java.base/java.lang.Integer.parseInt(Integer.java:668)
        at java.base/java.lang.Integer.parseInt(Integer.java:786)
        at com.oltpbenchmark.benchmarks.auctionmark.util.ItemId.decode(ItemId.java:62)
        at com.oltpbenchmark.benchmarks.auctionmark.util.ItemId.<init>(ItemId.java:50)
        at com.oltpbenchmark.benchmarks.auctionmark.util.AuctionMarkUtil.getUniqueElementId(AuctionMarkUtil.java:38)
        at com.oltpbenchmark.benchmarks.auctionmark.procedures.NewItem.run(NewItem.java:203)
        at com.oltpbenchmark.benchmarks.auctionmark.AuctionMarkWorker.executeNewItem(AuctionMarkWorker.java:748)
        at com.oltpbenchmark.benchmarks.auctionmark.AuctionMarkWorker.executeWork(AuctionMarkWorker.java:371)
        at com.oltpbenchmark.api.Worker.doWork(Worker.java:425)
        at com.oltpbenchmark.api.Worker.run(Worker.java:287)
        at java.base/java.lang.Thread.run(Thread.java:833)
``` 

Am i missing something here? Any pointers are greatly appreciated.

**Comment by apavlo** [comment]
@pct960 What scalefactor?

**Comment by pct960** [comment]
Hi Andy,

I was using a scalefactor of 1. Also, the warnings about duplicate items are especially persistent. There are hundreds of those thrown on every run.

Another issue I noticed with AuctionMark is that transactions are wildly varying in latencies. NewBid, for instance, ranged anywhere between 2 ms to 20,000 ms with spikes appearing throughout the timeline. I thought it maybe had something to do with batched writes, but when I turned it off, the issue still persisted.

Any idea how I can get around these issues? Thanks.
